src/context/filter.rs

```diff
@@ -0,0 +1,99 @@
+    RedisModule_CommandFilterArgDelete, RedisModule_CommandFilterArgGet,
+    RedisModule_CommandFilterArgInsert, RedisModule_CommandFilterArgReplace,
+    RedisModule_CommandFilterArgsCount, RedisModule_CommandFilterGetClientId,
+    RedisModule_CreateString, RedisModule_RegisterCommandFilter,
+    RedisModule_UnregisterCommandFilter, ValkeyString,
+};
+use std::ffi::{c_int, CString};
+use std::ptr::null_mut;
+
+#[derive(Debug)]
+pub struct CommandFilter {
+    pub inner: *mut RedisModuleCommandFilter,
+}
+
+// otherwise you get error:     cannot be shared between threads safely
+unsafe impl Send for CommandFilter {}
+unsafe impl Sync for CommandFilter {}
+
+impl Context {
+    pub fn register_command_filter(
+        &self,
+        module_cmd_filter_func: extern "C" fn(*mut RedisModuleCommandFilterCtx),
+        flags: c_int,
+    ) -> CommandFilter {
+        let module_cmd_filter = unsafe {
+            RedisModule_RegisterCommandFilter.unwrap()(
+                self.ctx,
+                Some(module_cmd_filter_func),
+                flags,
+            )
+        };
+        CommandFilter {
+            inner: module_cmd_filter,
+        }
+    }
+    pub fn unregister_command_filter(&self, cmd_filter: &CommandFilter) {
+        unsafe {
+            RedisModule_UnregisterCommandFilter.unwrap()(self.ctx, cmd_filter.inner);
+        }
+    }
+}
+
+pub fn command_filter_args_count(ctx: *mut RedisModuleCommandFilterCtx) -> c_int {
+    unsafe { RedisModule_CommandFilterArgsCount.unwrap()(ctx) }
+}
+
+pub fn command_filter_arg_get(
+    ctx: *mut RedisModuleCommandFilterCtx,
+    pos: c_int,
+) -> *mut RedisModuleString {
+    unsafe { RedisModule_CommandFilterArgGet.unwrap()(ctx, pos) }
+}
+
+/// wrapper to get argument as a String instead of RedisModuleString
+pub fn command_filter_arg_get_as_string(
+    ctx: *mut RedisModuleCommandFilterCtx,
+    pos: c_int,
+) -> String {
+    let arg = command_filter_arg_get(ctx, pos);
+    ValkeyString::from_ptr(arg).unwrap().to_string()
+}
+
+pub fn command_filter_arg_replace(
+    ctx: *mut RedisModuleCommandFilterCtx,
+    pos: c_int,
+    arg: *mut RedisModuleString,
+) {
+    unsafe { RedisModule_CommandFilterArgReplace.unwrap()(ctx, pos, arg) };
+}
+
+pub fn command_filter_arg_insert(
+    ctx: *mut RedisModuleCommandFilterCtx,
+    pos: c_int,
+    arg: *mut RedisModuleString,
+) {
+    unsafe { RedisModule_CommandFilterArgInsert.unwrap()(ctx, pos, arg) };
+}
+
+pub fn command_filter_arg_delete(ctx: *mut RedisModuleCommandFilterCtx, pos: c_int) {
+    unsafe { RedisModule_CommandFilterArgDelete.unwrap()(ctx, pos) };
+}
+
+pub fn command_filter_get_client_id(ctx: *mut RedisModuleCommandFilterCtx) -> u64 {
+    unsafe { RedisModule_CommandFilterGetClientId.unwrap()(ctx) }
+}
+/// create a RedisModuleString from a &str without Context which is not present in filter functions
+pub fn arg_module_create_string(arg: &str) -> *mut RedisModuleString {
```

Comment:
It is possible to do this today with the existing `ValkeyString` interface / structure by providing a Dummy Context.

You can do this with either of the APIs below
* https://github.com/valkey-io/valkeymodule-rs/blob/c68bc202ecb9fe78a515f5d7912cb5a39ee24e2f/src/redismodule.rs#L154
* https://github.com/valkey-io/valkeymodule-rs/blob/c68bc202ecb9fe78a515f5d7912cb5a39ee24e2f/src/redismodule.rs#L165

Reply:
thank you, this is so much easier `Context::dummy().create_string("info2").inner`